src/suricata.c

```diff
@@ -2369,6 +2369,7 @@ static int FinalizeRunMode(SCInstance *suri, char **argv)
 {
     switch (suri->run_mode) {
         case RUNMODE_UNKNOWN:
+            SCLogError("Missing run mode. Please specify a capture runmode.");
```

Comment:
I'm thinking this could be a one liner. Maybe @lukashino has a better idea of what one might want to see in error message in such a case.

Reply:
Okay, I wait for his feedback

Reply:
Considering the switch statement is empty I would suggest a transformation to the if-clause (without else part, just asking for the unknown runmode)

I also feel like the word "runmode" is overloaded in Suricata as it is used for thread runmodes (single, autofp, workers) and capture modes (af_packet, dpdk, etc.). So I would suggest only referring to it as a capture mode. I will create a new ticket that would rename "runmode" to e.g. capture mode in places where it actually refers to capture mode. In the meantime please edit the error message to something like "no capture mode specified as a command-line argument, specify one from the --list-runmodes"

Reply:
Okay, thanks for the feedback